### PR TITLE
fix(a11y): fixing MultiSelect no focus indicator

### LINKED ISSUE #6855: MultiSelect - accessibility issue - no focus indicator visible on clear button
1. go to https://react.carbondesignsystem.com/?path=/story/multiselect--default
2. select a few elements
3. When we focus on the button to "Clear Selection" by keyboard, there is no focus indicator visible.

Accessibility checkpoint:  2.4.7 Focus Visible
（https://www.ibm.com/able/guidelines/ci162/focus_visible.html）

see screenshot:
image

### PR DESCRIPTION
**Pull request by rjhenriquez**
Closes #6855

Fixing MultiSelect - accessibility issue - no focus indicator visible on clear button

#### Changelog

**Changed**

MultiSelect component

#### Testing / Reviewing

- Go to http://localhost:9000/?path=/story/multiselect--default
- Select a few elements
- Focus on the button to "Clear Selection" by keyboard, it should show an outline

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

**Review by joshblack — COMMENTED** [review]
Actually, taking a look at the history for `_tag.scss` will this create a regression of: https://github.com/carbon-design-system/carbon/pull/4792 potentially?

**Comment by tw15egan** [comment]
@joshblack it seems like that regression already exists. Perhaps we need to add a `pointer-events: none` to the disabled tag?

Sep-28-2020 15-22-13

**Review by aledavila — APPROVED** [review]

**Review by aledavila — CHANGES_REQUESTED** [review]
Ah now I see it. @rjhenriquez did you want to look at the possible fix TJ suggested ?

**Comment by rjhenriquez** [comment]
@aledavila @tw15egan `pointer-events: none;` added

**Comment by tw15egan** [comment]
I'm not seeing the disabled focus anymore, but it seems like the rest of the changes are gone now? Not able to tab to the clear icon

**Comment by joshblack** [comment]
bump @rjhenriquez when you get a sec to respond to above 👀

**Comment by tw15egan** [comment]
Just a few small issues I'm still seeing:

- [x] Filter tag no longer receives `not-allowed` cursor styles when disabled. We may need to handle the onClick behavior in the React code and just return if disabled is true




- [x] Multiselect badge does not receive any disabled styles when the multiselect is disabled (existing bug, can be tackled in a separate PR)
Screen Shot 2020-10-28 at 11 28 27 AM

- [x] Multiselect badge loses focus styles when hovered over 

multiselect

**Comment by rjhenriquez** [comment]
@tw15egan 

- Sadly, `pointer-events: none` will override `cursor: not-allowed`, making the cursor default to the normal one. Do we create another issue to handle the onClick behavior in the React code ?

- > Multiselect badge does not receive any disabled styles when the multiselect is disabled
I think it's now fixed

- > Multiselect badge loses focus styles when hovered over
The `.bx--tag--filter:hover` has no outline on hover, so I added a hover inside the focus state.. weird but it works.

File: packages/components/src/components/list-box/_list-box.scss [hunk 1]
```diff
@@ -498,13 +502,15 @@ $list-box-menu-width: rem(300px);
     }
   }
 
-  .#{$prefix}--list-box--disabled
-    .#{$prefix}--list-box__selection--multi
+  .#{$prefix}--list-box--disabled .#{$prefix}--list-box__selection--multi {
+    @include tag-theme($disabled-02, $disabled-01);
```
Comment (tw15egan) [inline packages/components/src/components/list-box/_list-box.scss:507]: Seems like `tag-theme` is not imported in `_list-box.scss` and is causing some build errors. 

We'll need to add a `@import '../tag/mixins';` to the top of `_list-box.scss`

**Comment by tw15egan** [comment]
@rjhenriquez I've gone ahead and updated this PR so we can get it out before the release tomorrow 👍

**Review by aledavila — APPROVED** [review]

**Review by emyarod — APPROVED** [review]
thanks for the updates @tw15egan, looks good to me

**Pull request merged**